test/e2e/reconciler/cache/replication_test.go

```diff
@@ -144,225 +163,195 @@ func replicateAPIResourceSchemaNegativeScenario(ctx context.Context, t *testing.
-	})
-	scenario.VerifyReplication(ctx, t, clusterName.Path())
+// replicateWorkspaceTypeScenario tests if a WorkspaceType is propagated to the cache server.
+// The test exercises creation, modification and removal of the Shard object.
+func replicateWorkspaceTypeScenario(ctx context.Context, t *testing.T, server framework.RunningServer, kcpShardClusterDynamicClient kcpdynamic.ClusterInterface, cacheKcpClusterDynamicClient kcpdynamic.ClusterInterface) {
+	t.Helper()
+	replicateResource(ctx,
+		t,
+		server,
+		kcpShardClusterDynamicClient,
+		cacheKcpClusterDynamicClient,
+		"",
+		"replicate-workspace-type",
+		"WorkspaceType",
+		tenancyv1alpha1.SchemeGroupVersion.WithResource("workspacetypes"),
+		&tenancyv1alpha1.WorkspaceType{ObjectMeta: metav1.ObjectMeta{Name: "replicate-workspace-type"}, Spec: tenancyv1alpha1.WorkspaceTypeSpec{}},
+		&tenancyv1alpha1.WorkspaceType{Spec: tenancyv1alpha1.WorkspaceTypeSpec{AdditionalWorkspaceLabels: map[string]string{"foo": "bar"}}},
+	)
+}
 
-	t.Logf("Verify that deleting source Shard %s/%s leads to removal of the cached object", clusterName, resourceName)
-	scenario.DeleteSourceResourceAndVerify(ctx, t, clusterName.Path())
+// replicateWorkspaceTypeNegativeScenario checks if modified or even deleted cached WorkspaceType will be reconciled to match the original object.
+func replicateWorkspaceTypeNegativeScenario(ctx context.Context, t *testing.T, server framework.RunningServer, kcpShardClusterDynamicClient kcpdynamic.ClusterInterface, cacheKcpClusterDynamicClient kcpdynamic.ClusterInterface) {
+	t.Helper()
+	replicateResourceNegative(
+		ctx,
+		t,
+		server,
+		kcpShardClusterDynamicClient,
+		cacheKcpClusterDynamicClient,
+		"",
+		"replicate-workspace-type-negative",
+		"WorkspaceType",
+		tenancyv1alpha1.SchemeGroupVersion.WithResource("workspacetypes"),
+		&tenancyv1alpha1.WorkspaceType{ObjectMeta: metav1.ObjectMeta{Name: "replicate-workspace-type-negative"}, Spec: tenancyv1alpha1.WorkspaceTypeSpec{}},
+		&tenancyv1alpha1.WorkspaceType{Spec: tenancyv1alpha1.WorkspaceTypeSpec{AdditionalWorkspaceLabels: map[string]string{"foo": "bar"}}},
+	)
 }
 
-// replicateShardNegativeScenario checks if modified or even deleted cached Shard will be reconciled to match the original object.
-func replicateShardNegativeScenario(ctx context.Context, t *testing.T, server framework.RunningServer, kcpShardClusterClient kcpclientset.ClusterInterface, cacheKcpClusterClient kcpclientset.ClusterInterface) {
+// replicateResource tests if the given resource is propagated to the cache server.
+// The test exercises creation, modification and removal of the resource.
+func replicateResource(ctx context.Context, t *testing.T,
+	server framework.RunningServer, kcpShardClusterDynamicClient kcpdynamic.ClusterInterface, cacheKcpClusterDynamicClient kcpdynamic.ClusterInterface,
+	clusterName logicalcluster.Name, resourceName string, kind string, gvr schema.GroupVersionResource, res runtime.Object, resWithModifiedSpec runtime.Object) {
```

Comment:
yes, an alternative would be to provide a struct but filling all required fields would at the calling line look the same (almost) as providing params. I will try to think about how it could be changed.